extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,239 @@
+
+define([
+    'jquery',
+    './common',
+], function ($, common) {
+    "use strict";
+
+    var updatepkg = {
+
+        update: function (packages, project) {
+            let payload = {};
+            payload['project'] = project;
+            payload['packages'] = packages;
+            let settings = {
+                "async": true,
+                "crossDomain": true,
+                "url": "http://localhost:8888/api/packagemanager/packages",
+                "method": "PATCH",
+                "headers": {
+                    "Content-Type": "application/json",
+                    "cache-control": "no-cache",
+                },
+                "processData": false,
+                "data": JSON.stringify(payload)
+            };
+
+
+            return new Promise(resolve => {
+                $.ajax(settings).done(function (response) {
+                    resolve(response);
+                });
+            });
+        },
+
+        checkupdate: function (project) {
+            var settings = {
+                "async": true,
+                "crossDomain": true,
+                "url": "http://localhost:8888/api/packagemanager/packages/check_update?project=" + project,
+                "method": "GET",
+                "headers": {
+                    "Content-Type": "application/json",
+                    "cache-control": "no-cache"
+                },
+                "processData": false,
+                "data": ""
+            };
+
+            return new Promise(resolve => {
+                $.ajax(settings).done(function (response) {
+                    resolve(response);
+                });
+            });
+        },
+
+        checknewversion: function (updates, packagename, pkgversion) {
+            for (let element of updates) {
+                if (element.name === packagename) {
+                    return element.version;
+                }
+            }
+            return pkgversion;
+        },
+
+        load: async function () {
+            let update = this.update;
+            let checkupdate = this.checkupdate;
+            let checknewversion = this.checknewversion;
+            jQuery(async function () {
```

Comment:
you already have jQuery loaded in the var $, imported by requirejs. Why do you call jQuery directly?

Reply:
Also, I want to see the API calls separated from the rest.